src/github.com/getlantern/balancer/balancer.go

```diff
@@ -151,6 +139,25 @@ func (b *Balancer) checkDialers() {
 	b.mu.RUnlock()
 }
 
+func (b *Balancer) pickDialer(trustedOnly bool) (*dialer, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
```

Comment:
@oxtoacart I didn't change to `RLock` though the lock is indeed `RWMutex`. There are heap operations at the end.

I extracted the code to separate method to always unlock the mutex.